src/Mvc/Mvc.Core/src/ModelBinding/Binders/ServicesModelBinderProvider.cs

```diff
@@ -26,9 +25,16 @@ public class ServicesModelBinderProvider : IModelBinderProvider
         if (context.BindingInfo.BindingSource != null &&
             context.BindingInfo.BindingSource.CanAcceptDataFrom(BindingSource.Services))
         {
-            return _modelBinder;
+            return new ServicesModelBinder()
+            {
+                IsOptionalParameter = IsOptionalParameter(context.Metadata.Identity.ParameterInfo!)
```

Comment:
This wouldn't work for properties.

Reply:
I can easily update to check before but can properties be bound from services?